### ISSUE TITLE
Bad Company 2 - weird sky on EA servers only, conquest mode

### ISSUE THREAD

**Issue by Bumbadawg** [doitsujin/dxvk#1156]
Hi there !
I reported this issue on proton with log/screenshots:
https://github.com/ValveSoftware/Proton/issues/200#issuecomment-518643828

It's really specific. The game runs flawlessly at 140~200fps most of the time, kudos for the Vulkan layer mate.
But i get a weird blue sky on some maps (Atacama Desert and Laguna Presa) in "conquest mode" on EA servers only (where i could reproduce the issue). 

It has not happened outside EA servers on those maps/modes with Punkbuster enabled.

I tried enabling/disabling feral's gamemode, fsync (running Valve kernel 5), and changing graphics settings from high to low, no changes for this issue.

Not tightly related but iirc, ENB on GTA V back in the day had a similar bug where the sky went blue.

**Comment by doitsujin** [comment]
Can't debug without an apitrace.

**Comment by Bumbadawg** [comment]
Sure thing but as a trace neophyte, i couldn't get to produce an apitrace..

I'm using Proton from the Steam client. Launching with DXVK disabled as mentionned on some posts.
 I copy the DLLs in the game's folder as you indicated https://github.com/doitsujin/dxvk/wiki/Common-issues#apitrace but nothing gets generated. It didn't generate anything in the pfx/Desktop (which should be the place to produce the trace according to the doc).

So i tried a 2nd method. Renaming apitrace.exe to the game.exe and the game.exe to another. Steam launcher option: `PROTON_USE_WINED3D=1 %command% trace -a dxgi BFBC2Game_o.exe` (i also tried just with trace game.exe)

It required to have the lib/wrapper folder up the parent folder, which lead to this message  (in verbose):
```
info: did not find Z:\$HOME\.steam\steam\steamapps\common\Battlefield Bad Company 2\..\wrappers\RelWithDebInfo\dxgitrace.dll
info: found Z:\$HOME\.steam\steam\steamapps\common\Battlefield Bad Company 2\..\lib\wrappers\dxgitrace.dll
inject -D "Z:\$HOME\.steam\steam\steamapps\common\Battlefield Bad Company 2\..\lib\wrappers\dxgitrace.dll" -- BFBC2Game_o.exe
error: failed to execute inject
0009:fixme:kernelbase:AppPolicyGetProcessTerminationMethod FFFFFFFA, 0034FEB4
```
Any hint on what i did wrong to produce the trace ?

**Comment by doitsujin** [comment]
Have you copied all the `d3d11.dll`, `dxgi.dll` and `dxgitrace.dll` files of the correct architecture next to the game exe?

**Comment by Bumbadawg** [comment]
NB: the game is 32 bit d3d11.

So for the 1st method mentionned yes, i did: 
- copy from the build "apitrace-8.0.20190414-win32" the lib/wrappers/dxgi.* the d3d11.dll, 
- chmod +x the files (since they were not executable)
and start the game in Steam with `PROTON_USE_WINED3D=1 %command%`
i tried adding apitrace.exe as well, but nothing.
I'm trying to the find the trace with `find ~ -name *.trace`. I the trace should be output in `~/.steam/steam/steamapps/compatdata/24960/pfx/drive_c/users/steamuser/Desktop`
Maybe a global variable makes the .exe look elsewhere for its DLLs and hence doesn't load the dxgi.dll ?

**Comment by doitsujin** [comment]
Can you try without `PROTON_USE_WINED3D=1`? Might be fine in this case, but that option also sets `WINEDLLOVERRIDES` to use built-in DXGI and D3D11 which is the problem you're running into.

**Comment by Bumbadawg** [comment]
Indeed. So i had to manually add dxgi and d3d11 DLLs through winecfg in the pfx, then it loaded the DLLs. But it crashed. 

Now launching with DXVK (`PROTON_USE_WINED3D`): it starts. But black screen for 3mn and nothing impressive in proton log. Unresponsive window so i closed. Should have i kept it running for longer ? 

[steam-24960.log](https://github.com/doitsujin/dxvk/files/3474021/steam-24960.log)
[BFBC2Game.1.trace.tar.gz](https://github.com/doitsujin/dxvk/files/3474023/BFBC2Game.1.trace.tar.gz)

**Comment by Bumbadawg** [comment]
Game crashed with DXVK + apitrace but it's supposed not to work according to some post 
>>> DO NOT use DXVK together with apitrace!

 [steam-24960.log](https://github.com/doitsujin/dxvk/files/3474032/steam-24960.log)

**Comment by Blisto91** [comment]
Been banging my head against the screen trying to install any driver below 470 on this setup. But i kinda gave up on that so this isn't a proper test to reproduce the original issue.

With a nvidia 960 with 470.103 driver i have not been able to reproduce this issue.
I tried with the dxvk version that came along with proton 4.11 and manually replacing with 1.3.3, 1.3.4 and 1.10.
I have tried going between servers and seeing if i could reproduce the issue on the specific maps and with all setting level, but have not been able to.
